WatchDogCore/WatchDog/src/nl/tudelft/watchdog/core/logic/interval/intervaltypes/IntervalBase.java

```diff
@@ -122,7 +122,11 @@ public IntervalType getType() {
 
 	/** Necessary for storage of Intervals. */
 	public int compareTo(IntervalBase comparedInterval) {
-		return getEnd().compareTo(comparedInterval.getEnd());
+		int res = getEnd().compareTo(comparedInterval.getEnd());
+		if (res == 0 && !this.equals(comparedInterval)) {
+			res = -1;
+		}
+		return res;
```

Comment:
Ah right, I see your point now. The problem is that compareTo(a,b) was entirely defined in terms of compareTo(a.end,b.end), but with this implementation this is no longer true in all cases. So, what we need is that res is not always set to 1 in this case, but to -1 or 1 depending on some other property of the intervals. In that way, compareTo (b,a)=1 when compareTo(a,b)=-1 in case compareTo(a.end, b.end)=0. I'll see if i can find such a property soon.